spec/forecasts/forecast.js

```diff
@@ -1,9 +1,79 @@
+import {Forecast} from "../../src/forecasts/forecast.js";
+import * as weather from "weather-js";
+import {staticWeather} from "../static-resource/weather.json"; //Mock data
 
-describe("here1>", () => {
+describe("Forecast", () => {
+  let forecast;
 
-  it('will assert true to equal true', () => {
+  describe("getWeather", () => {
+    let expectedErrMsg;
 
-    expect(true).to.equal(true);
+    beforeEach(() => {
+      forecast = new Forecast();
+      expectedErrMsg = "Values undefined";
+    });
+
+    afterEach(() => {
+      forecast = null;
+    });
+
+    describe("with falsy params", () => {
+      it("Return early when either params undefined", () => {
+        forecast.getWeather(undefined, undefined, (err, result) => {
+          expect(err).to.equal(expectedErrMsg);
+        });
+      });
+
+      it("Return early when params are null", () => {
+          forecast.getWeather(null, null, (err, result) => {
+            expect(err).to.equal(expectedErrMsg);
+          });
+      });
+
+      it("Return early when params are 0", () => {
+        forecast.getWeather(0, 0, (err, result) => {
+          expect(err).to.equal(expectedErrMsg);
+        });
+      });
+
+      it("Return early when params are NaN", () => {
+        forecast.getWeather(NaN, NaN, (err, result) => {
+          expect(err).to.equal(expectedErrMsg);
+        });
+      });
+
+    });
+
+    describe("Weather.find callback result", () => {
+      let stub;
+
+      beforeEach(() => {
+        forecast = new Forecast();
+        stub = sinon.stub(weather, "find"); //Stub any callout to this method
+        expectedErrMsg = "CallbackError";
+      });
+
+      afterEach(() => {
+        forecast = null;
+        stub.restore();
+      });
+
+      it("Tests getWeather returns json object", () => {
+        stub.yields(null, staticWeather); //Mock test data
+        forecast.getWeather("London, England", "C", (err, result) => {
+          expect(result[0].name).to.equal(staticWeather[0].name);
+          expect(result[0].degreetype).to.equal(staticWeather[0].degreetype);
+        });
+      });
+
+      it("Tests getWeather callback throws an error", () => {
+        stub.yields(expectedErrMsg);
+        forecast.getWeather("London, England", "C", (err, result) => {
+          expect(err).to.equal(expectedErrMsg);
+        });
+      });
+
+    });
```

Comment:
Could add test after this to asser that weather.find is called with the right params passed into out function

Reply:
Not sure what you mean here? We check for false params and use correct ones for the stubs do you mean run another get weather test and check it doesn't return early?

Reply:
Sorry i mean do a new it block, and ensure what we passed in, is used when calling weather.find. so your assert will be checking weather.find it called with